Setup Webpack for React (using Babel) and CSS

van-hsieh opened the pull request.

van-hsieh committed:
Webpack, babel, css loader, style-loader, file-loader setup

van-hsieh committed:
with '#' will be ignored, and an empty message aborts the commit.

van-hsieh commented:
@fernanda1411 @cdeng1111 Please review the setup and once confirmed, merge the request. Thank you!

van-hsieh committed:
test if deals with commiting .gitignore

van-hsieh committed:
Delete .gitignore

fernanda1411 reviewed (changes requested):
Great work. I have some requests and suggestions:
 
a) Can we remove `node_modules` folder from this PR and add it to `.gitignore`? Right now your PR has over 5K files.  :relaxed:

Here is how you can do that:

1. On you local branch, delete the folder `node_modules` (that can be generate at any time by running `npm install`).
2. Add a `.gitignore` file to the project root folder and in it write this:

```jsx
node_modules/
```

3. Make a commit and then push it to  [`van-hsieh:main`](https://github.com/van-hsieh/sycamore) on your forked repo in github. That should update this Pull Request. 

b) On package.json let's change the "repository url" to the url of the organization repo: `https://github.com/sensational-sycamore/sycamore` 

c) On package.json let's change the "homepage" url to the organization repo: `"homepage": "https://github.com/sensational-sycamore/sycamore#readme"`

d) On the `webpack.config.js` I see that we haven't added support to `sass` yet. We can do that later on another PR if you prefer, but if you can let's look into that.

e) Also, I think we should start using branches on our computers instead of working on `main` / `master`.  Here is a great article that I found explaining this: 

[Github Forking Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow)

van-hsieh committed:
added .gitignore and ignored .env file

van-hsieh committed:
Added ESLint and cleaned up file

van-hsieh committed:
added node_modules to gitignore

van-hsieh committed:
updated package,json repo url and homepage, and added sass (+ sass config for webpack)

van-hsieh committed:
remove node_modules

van-hsieh commented:
@fernanda1411 You are such an amazing reviewer!!! Thanks for the feedback. 

I updated the package.json, deleted the node_modules, added node_modules to .gitignore, and set up the ESLint. Let me know if there's anything else I should change.

fernanda1411 commented on package.json:
```diff
@@ -0,0 +1,42 @@
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/van-hsieh/sycamore/issues"
```
Can you please update this url to the organization url too please? :)

fernanda1411 reviewed (changes requested):
I think you can remove `client/dist/main.js` and add it to the `.gitignore` too as this file is generated by webpack. :)

I also noticed that we are using the wrong github flow. I'll update my pull request https://github.com/sensational-sycamore/sycamore/pull/2. We should be using [Feature Branch Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) as explained on [LEARN](https://learn-2.galvanize.com/cohorts/3050/blocks/94/content_files/Front%20End%20Capstone/introduction-to-the-front-end-capstone.md). I think we can start doing that moving forward after we merge the open PRs.

fernanda1411 commented on .eslintignore:
```diff
@@ -0,0 +1,8 @@
+**/*.min.js
+client/data/*
+compiled/
+main.js
```
nit: can you add a blank line in the end of files to avoid the "no new line at end of file" github alert?

fernanda1411 commented on client/dist/index.html:
```diff
@@ -0,0 +1,13 @@
+  <div id="app"></div>
+  <script src="main.js"></script>
+</body>
+</html>
```
nit: can you add a blank line in the end of files to avoid the "no new line at end of file" github alert?

fernanda1411 commented on client/src/App.jsx:
```diff
@@ -0,0 +1,7 @@
+  <div>
+    <h1>Hello World</h1>
+  </div>;
+};
```
nit: can you add a blank line in the end of files to avoid the "no new line at end of file" github alert?

fernanda1411 commented on client/src/index.js:
```diff
@@ -0,0 +1,8 @@
+ReactDOM.render(
+  <div>Hello World</div>,
+  document.getElementById('app')
+);
```
nit: can you add a blank line in the end of files to avoid the "no new line at end of file" github alert?

van-hsieh replied:
Thanks Nanda! Just resolved all listed issues and pushed.

fernanda1411 commented on webpack.config.js:
```diff
@@ -0,0 +1,38 @@
+  resolve: {
+    extensions: ['*', '.js', '.jsx'],
+  }
+};
```
nit: can you add a blank line in the end of files to avoid the "no new line at end of file" github alert?

van-hsieh committed:
added client/dist/main.js to .gitignore

van-hsieh committed:
removed client/dist/main.js from commit

van-hsieh committed:
updated bugs url in package.json

van-hsieh committed:
added new line in end of file to avoid - no new line at end of file github alert

van-hsieh committed:
added new line in end of file to avoid - no new line at end of file github alert

fernanda1411 commented on .gitignore:
```diff
@@ -0,0 +1,5 @@
+node_modules/
+.DS_Store
+compiled/
+client/dist/main.js
```
sorry :) can we add a blank line to the end of this file too?

van-hsieh replied:
All set!

van-hsieh committed:
added blank line after .gitignore

van-hsieh committed:
added new line to end of webpack.config.js
